### PR TITLE
Work through webpage 1 to 2 migration steps

### PR DESCRIPTION
**Pull request by steckel**
I'm not entirely sure what the status of moving to webpack 2 is (I did see #637 but it's just a simple dependency change at the moment).

I went through the exercise of going through the migration from webpack 1 to webpack 2 steps and attempted to run the demo project after linking. Things seem to work.

I'm happy to iterate on this PR if this is work that would be of value for upstream.

### CODE REVIEW

File: dist/server/config.js [hunk 1]
```diff
@@ -12,13 +12,11 @@ var _extends2 = require('babel-runtime/helpers/extends');
...
 
-exports.addJsonLoaderIfNotAvailable = addJsonLoaderIfNotAvailable;
-
 exports.default = function (configType, baseConfig, configDir) {
   var config = baseConfig;
 
-  var babelConfig = (0, _babel_config2.default)(configDir);
-  config.module.loaders[0].query = babelConfig;
+  // const babelConfig = loadBabelConfig(configDir);
+  // config.module.rules[0].query = babelConfig;
```
Comment (steckel) [inline dist/server/config.js]: [ ] re-implement bableConfig

File: dist/server/config/defaults/webpack.config.js [hunk 2]
```diff
@@ -19,36 +19,43 @@ function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { de
...
+  //     autoprefixer({
+  //       browsers: [
+  //         '>1%',
+  //         'last 4 versions',
+  //         'Firefox ESR',
+  //         'not ie < 9',
+  //       ],
+  //     }),
+  //   ];
+  // };
```
Comment (steckel) [inline dist/server/config/defaults/webpack.config.js]: - [ ] reimplement postcss options

File: dist/server/config/webpack.config.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ Object.defineProperty(exports, "__esModule", {
 
 exports.default = function () {
   var config = {
-    devtool: 'eval',
+    devtool: 'source-map',
```
Comment (steckel) [inline dist/server/config/webpack.config.js]: - [ ] revert devtool edit

File: dist/server/config/webpack.config.js [hunk 4]
```diff
@@ -18,21 +18,23 @@ exports.default = function () {
...
     },
     resolve: {
       // Since we ship with json-loader always, it's better to move extensions to here
       // from the default config.
-      extensions: ['.js', '.json', '.jsx', ''],
+      extensions: ['.js', '.json', '.jsx'],
       // Add support to NODE_PATH. With this we could avoid relative path imports.
       // Based on this CRA feature: https://github.com/facebookincubator/create-react-app/issues/253
-      fallback: _utils.nodePaths,
+      // fallback: nodePaths,
```
Comment (steckel) [inline dist/server/config/webpack.config.js]: - [ ] investigate if fallback was preserved in another way in webpack 2

File: dist/server/config/webpack.config.prod.js [hunk 5]
```diff
@@ -35,21 +35,23 @@ exports.default = function () {
...
     },
     resolve: {
       // Since we ship with json-loader always, it's better to move extensions to here
       // from the default config.
-      extensions: ['.js', '.json', '.jsx', ''],
+      extensions: ['.js', '.json', '.jsx'],
       // Add support to NODE_PATH. With this we could avoid relative path imports.
       // Based on this CRA feature: https://github.com/facebookincubator/create-react-app/issues/253
-      fallback: _utils.nodePaths,
+      // fallback: nodePaths,
```
Comment (steckel) [inline dist/server/config/webpack.config.prod.js]: ditto

File: src/server/config.js [hunk 6]
```diff
@@ -8,41 +8,14 @@ import { includePaths } from './config/utils';
...
 // `baseConfig` is a webpack configuration bundled with storybook.
 // React Storybook will look in the `configDir` directory
 // (inside working directory) if a config path is not provided.
 export default function (configType, baseConfig, configDir) {
   const config = baseConfig;
 
-  const babelConfig = loadBabelConfig(configDir);
-  config.module.loaders[0].query = babelConfig;
+  // const babelConfig = loadBabelConfig(configDir);
+  // config.module.rules[0].query = babelConfig;
```
Comment (steckel) [inline src/server/config.js]: ditto

File: src/server/config/defaults/webpack.config.js [hunk 7]
```diff
@@ -4,53 +4,54 @@ import { includePaths } from '../utils';
...
+  //     autoprefixer({
+  //       browsers: [
+  //         '>1%',
+  //         'last 4 versions',
+  //         'Firefox ESR',
+  //         'not ie < 9',
+  //       ],
+  //     }),
+  //   ];
+  // };
```
Comment (steckel) [inline src/server/config/defaults/webpack.config.js]: ditto

File: src/server/config/webpack.config.js [hunk 8]
```diff
@@ -14,7 +14,7 @@ import babelLoaderConfig from './babel.js';
 
 export default function () {
   const config = {
-    devtool: 'eval',
+    devtool: 'source-map',
```
Comment (steckel) [inline src/server/config/webpack.config.js]: ditto

File: src/server/config/webpack.config.js [hunk 9]
```diff
@@ -39,23 +39,25 @@ export default function () {
...
     },
     resolve: {
       // Since we ship with json-loader always, it's better to move extensions to here
       // from the default config.
-      extensions: ['.js', '.json', '.jsx', ''],
+      extensions: ['.js', '.json', '.jsx'],
       // Add support to NODE_PATH. With this we could avoid relative path imports.
       // Based on this CRA feature: https://github.com/facebookincubator/create-react-app/issues/253
-      fallback: nodePaths,
+      // fallback: nodePaths,
```
Comment (steckel) [inline src/server/config/webpack.config.js]: ditto

File: src/server/config/webpack.config.prod.js [hunk 10]
```diff
@@ -50,23 +49,25 @@ export default function () {
...
     },
     resolve: {
       // Since we ship with json-loader always, it's better to move extensions to here
       // from the default config.
-      extensions: ['.js', '.json', '.jsx', ''],
+      extensions: ['.js', '.json', '.jsx'],
       // Add support to NODE_PATH. With this we could avoid relative path imports.
       // Based on this CRA feature: https://github.com/facebookincubator/create-react-app/issues/253
-      fallback: nodePaths,
+      // fallback: nodePaths,
```
Comment (steckel) [inline src/server/config/webpack.config.prod.js]: ditto

**Comment by mnquintana** [comment]
@steckel 👋  Are you still planning on finishing this up? Just wanted to check 😄

**Comment by steckel** [comment]
I have not done any additional work on this but I would be happy to finish it up if any of the maintainers mentioned that it would be something they would consider (cc @arunoda).

**Comment by jetpacmonkey** [comment]
The lack of Webpack 2 compatibility has made it extremely difficult to use storybooks. Hopefully this can be either finished/merged or forked soon, because maintaining two sets of webpack configs is a pain.

**Comment by ndelangen** [comment]
I hear you @jetpacmonkey and everyone subscribed! I'm working on this and there's lots of progress!

Thank you @steckel for taking the time to analyse the changed that needed to be made.
If you'd like to review the work I did here, that'd be appreciated!
https://github.com/storybooks/mono/commit/06fe5d7ee6be29ef882e0f51336257742708e0cd

It's already merged to that master 😉

**Comment by ndelangen** [comment]
So don't feel bad or like this isn't going to happen, because it's going to happen SOON!

I'm closing these PR because they will not be merged, because I'm doing them all in 1 swoop.

**Pull request closed without merging**